src/vs/editor/contrib/unusualLineTerminators/browser/unusualLineTerminators.ts

```diff
@@ -85,13 +86,25 @@ export class UnusualLineTerminatorsDetector extends Disposable implements IEdito
 			return;
 		}
 
-		const result = await this._dialogService.confirm({
-			title: nls.localize('unusualLineTerminators.title', "Unusual Line Terminators"),
-			message: nls.localize('unusualLineTerminators.message', "Detected unusual line terminators"),
-			detail: nls.localize('unusualLineTerminators.detail', "The file '{0}' contains one or more unusual line terminator characters, like Line Separator (LS) or Paragraph Separator (PS).\n\nIt is recommended to remove them from the file. This can be configured via `editor.unusualLineTerminators`.", basename(model.uri)),
-			primaryButton: nls.localize('unusualLineTerminators.fix', "Remove Unusual Line Terminators"),
-			secondaryButton: nls.localize('unusualLineTerminators.ignore', "Ignore")
-		});
+		if (this._isPresentingDialog) {
+			// we're currently showing the dialog, which is async.
+			// avoid spamming the user
+			return;
+		}
+
+		let result: IConfirmationResult;
+		try {
+			this._isPresentingDialog = true;
+			result = await this._dialogService.confirm({
```

Comment:
I'm surprised to see the dialog service throw. When does that happen?

Reply:
Always when I have a field that I use to avoid recursion, I use a try/finally block to make sure the field gets reset. This is such that the code is always correct regardless of where execution ends up.